src/PaymentGateways/Gateways/PayPalStandard/Webhooks/Listeners/PaymentUpdated.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Give\PaymentGateways\Gateways\PayPalStandard\Webhooks\Listeners;
+
+use Give\Helpers\Call;
+use Give\PaymentGateways\Gateways\PayPalStandard\Actions\ProcessIpnDonationRefund;
+use Give_Payment;
+
+/**
+ * Handle web_accept and cart transaction types.
+ * read more: https://developer.paypal.com/api/nvp-soap/ipn/IPNandPDTVariables/#link-ipntransactiontypes
+ *
+ * @unreleased
+ */
+class PaymentUpdated implements EventListener
+{
+
+    /**
+     * @inheritDoc
+     */
+    public function processEvent($eventData)
+    {
+        // Collect donation payment details.
+        $donation = new Give_Payment($eventData->custom);
+        $donationStatus = strtolower($eventData->payment_status);
+
+        switch (true) {
```

Comment:
I think this is a misuse of the `switch` statement. Instead, a series of if statements with early returns is fine.

Reply:
fixed: https://github.com/impress-org/givewp/compare/9a60c9c123a8...fb16e5a9f7c1